src/components/subtitlesync/subtitlesync.js

```diff
@@ -151,6 +151,14 @@ class SubtitleSync {
             subtitleSyncContainer.classList.add('hide');
         }
     }
+
+    update(updateValue) {
+        SubtitleSync.prototype.toggle();
+
+        const value = new Intl.NumberFormat({ maximumFractionDigits: 1 })
+            .format(updateValue(parseFloat(subtitleSyncSlider.value)));
```

Comment:
I used NumberFormat to truncate to 1 decimal digit, because when increasing/decreasing by 0.1s it ends up with many digits due to being float number. (see screenshot) I found `Number.toFixed` that does the same and is part of Safari [1] (and I assume webkit) since 2007, so I assume it could work on the older supported platforms and that is already used in multiple part of the codebase [2].

![Screenshot_20240531_231001](https://github.com/jellyfin/jellyfin-web/assets/505826/236a237d-c1da-4d40-a2ae-0e5a22321afd)
 
[1] https://caniuse.com/?search=toFixed
[2] https://github.com/search?q=repo%3Ajellyfin%2Fjellyfin-web%20toFixed&type=code

Reply:
If you have encountered the problem of floating points representation with your keyboard stepping code, technically it will also occur with the implemented keyboard dragging (TV display mode).

Formatting/trimming should be performed in:
https://github.com/jellyfin/jellyfin-web/blob/6633a7090bbcbdcf263b8f36db3f5b8bf703da8e/src/components/subtitlesync/subtitlesync.js#L34
https://github.com/jellyfin/jellyfin-web/blob/6633a7090bbcbdcf263b8f36db3f5b8bf703da8e/src/components/subtitlesync/subtitlesync.js#L94

Reply:
when dragging `value` seems to be read directly from the html element [1], so it is already a string and does not cause float representation issues, as that fomatting is already managed by the browser. While my PR does the increment/decrement on float numbers directly, and can get in the issue.

[1] https://github.com/jellyfin/jellyfin-web/blob/6633a7090bbcbdcf263b8f36db3f5b8bf703da8e/src/components/subtitlesync/subtitlesync.js#L76

Reply:
> While my PR does the increment/decrement on float numbers directly

Your value goes to the `slider.value` first.
https://github.com/jellyfin/jellyfin-web/blob/6633a7090bbcbdcf263b8f36db3f5b8bf703da8e/src/components/subtitlesync/subtitlesync.js#L85
Then the code you are referring to:
https://github.com/jellyfin/jellyfin-web/blob/6633a7090bbcbdcf263b8f36db3f5b8bf703da8e/src/components/subtitlesync/subtitlesync.js#L76
and finally (where `offset` is a parsed value):
https://github.com/jellyfin/jellyfin-web/blob/6633a7090bbcbdcf263b8f36db3f5b8bf703da8e/src/components/subtitlesync/subtitlesync.js#L34

Keyboard dragging (not native, but our implementation for TV display mode) does the same:
https://github.com/jellyfin/jellyfin-web/blob/3ab1fbb1e4751fe86d65cee87b95f830c0fa77fb/src/elements/emby-slider/emby-slider.js#L468
Sends `input` event:
https://github.com/jellyfin/jellyfin-web/blob/3ab1fbb1e4751fe86d65cee87b95f830c0fa77fb/src/elements/emby-slider/emby-slider.js#L470-L474
The result value is then parsed and passed to the bubble generator:
_Btw, we probably should parse the value before percent computation (previous line)._
https://github.com/jellyfin/jellyfin-web/blob/3ab1fbb1e4751fe86d65cee87b95f830c0fa77fb/src/elements/emby-slider/emby-slider.js#L302
and finally (where `value` is a parsed value - a number already, so `parseFloat` is redundant, imo):
https://github.com/jellyfin/jellyfin-web/blob/6633a7090bbcbdcf263b8f36db3f5b8bf703da8e/src/components/subtitlesync/subtitlesync.js#L94

Reply:
Tested keyboard dragging (TV display mode). In Firefox, it has the same problem with interpretation of float values (near +-0.2, +-0.3). Chrome seems to snap the value when assigning `slider.value` (_Why doesn't Firefox do the same?_ :confused:).

I see 2 options:
1. Add public `snapValue` function to `emby-slider` and call it when necessary ([here](https://github.com/jellyfin/jellyfin-web/blob/3ab1fbb1e4751fe86d65cee87b95f830c0fa77fb/src/elements/emby-slider/emby-slider.js#L468) and [here](https://github.com/jellyfin/jellyfin-web/blob/6633a7090bbcbdcf263b8f36db3f5b8bf703da8e/src/components/subtitlesync/subtitlesync.js#L158C23-L158C34)).
2. Add private `snapValue` function to `emby-slider` and call it from [`valueset` event handler](https://github.com/jellyfin/jellyfin-web/blob/f950aaf78472911e7907e655f6eabbb5a7123de0/src/elements/emby-slider/emby-slider.js#L120) to fix the value (like in Chrome).

I've already implemented the mentioned `snapValue` function. All that remains is to decide which option we choose.

Reply:
I would vote for the second option.

Reply:
> I would vote for the second option.

#5667

Reply:
@carlo-colombo Now that #5667 has been merged and backported, you can remove `toFixed`, allowing the slider to trim the value accordingly with its step.